src/Physics/HadronTransport/INukeHadroData2025.cxx

```diff
@@ -0,0 +1,1900 @@
+  delete fXSecPipd_Abs;
+
+  delete fXSecPp_Cmp; //added to fix memory leak; no noticeable effect, but good convention.
+  delete fXSecPn_Cmp;
+  delete fXSecNn_Cmp;
+  delete fXSecPp_Tot;
+  delete fXSecPp_Elas;
+  delete fXSecPp_Reac;
+  delete fXSecPn_Tot;
+  delete fXSecPn_Elas;
+  delete fXSecPn_Reac;
+  delete fXSecNn_Tot;
+  delete fXSecNn_Elas;
+  delete fXSecNn_Reac;
+
+  // pi0n/p hA x-section splines
+  delete fXSecPi0n_Tot;
+  delete fXSecPi0n_CEx;
+  delete fXSecPi0n_Elas;
+  delete fXSecPi0n_Reac;
+  delete fXSecPi0p_Tot;
+  delete fXSecPi0p_CEx;
+  delete fXSecPi0p_Elas;
+  delete fXSecPi0p_Reac;
+  delete fXSecPi0d_Abs;
+
+  // K+N x-section splines
+  delete fXSecKpn_Elas;
+  delete fXSecKpp_Elas;
+  delete fXSecKpn_CEx;
+  delete fXSecKpN_Abs;
+  delete fXSecKpN_Tot;
+
+  // gamma x-section splines (inelastic only)
+  delete fXSecGamp_fs;
+  delete fXSecGamn_fs;
+  delete fXSecGamN_Tot;
+
+  // N+A x-section splines
+  delete fFracPA_Tot;
+  //  delete fFracPA_Elas;
+  delete fFracPA_Inel;
+  delete fFracPA_CEx;
+  delete fFracPA_Abs;
+  delete fFracPA_PiPro;
+  delete fFracNA_Tot;
+  //  delete fFracNA_Elas;
+  delete fFracNA_Inel;
+  delete fFracNA_CEx;
+  delete fFracNA_Abs;
+  delete fFracNA_PiPro;
+
+  delete fFracPA_Cmp;  //  cmp - add support later
+  delete fFracNA_Cmp;
+
+  // hN data
+  delete fhN2dXSecPP_Elas;
+  delete fhN2dXSecNP_Elas;
+  delete fhN2dXSecPipN_Elas;
+  delete fhN2dXSecPi0N_Elas;
+  delete fhN2dXSecPimN_Elas;
+  delete fhN2dXSecKpN_Elas;
+  delete fhN2dXSecKpP_Elas;
+  delete fhN2dXSecPiN_CEx;
+  delete fhN2dXSecPiN_Abs;
+  delete fhN2dXSecGamPi0P_Inelas;
+  delete fhN2dXSecGamPi0N_Inelas;
+  delete fhN2dXSecGamPipN_Inelas;
+  delete fhN2dXSecGamPimP_Inelas;
+  delete fhN2dXSecKpN_CEx;
+  
+ // delete TPipA_Tot;
+  delete TfracPipA_Abs;
+  delete TfracPipA_CEx;
+//  delete TfracPipA_Elas;
+  delete TfracPipA_Inelas;
+  delete TfracPipA_PiPro;
+    
+    delete TPipA_Tot;
+    delete TPipA_Abs;
+    delete TPipA_CEx;
+    delete TPipA_Inelas;
+    delete TPipA_PiPro;
+    
+  
+  // K+A x-section fraction splines
+  delete fFracKA_Tot;
+  delete fFracKA_Elas;
+  delete fFracKA_CEx;
+  delete fFracKA_Inel;
+  delete fFracKA_Abs;
+  
+}
+//____________________________________________________________________________
+INukeHadroData2025 * INukeHadroData2025::Instance()
+{
+  if(fInstance == 0) {
+    LOG("INukeData", pINFO) << "INukeHadroData2025 late initialization";
+    static INukeHadroData2025::Cleaner cleaner;
+    cleaner.DummyMethodAndSilentCompiler();
+    fInstance = new INukeHadroData2025;
+  }
+  return fInstance;
+}
+//____________________________________________________________________________
+void INukeHadroData2025::LoadCrossSections(void)
+{
+// Loads hadronic x-section data
+
+  //-- Get the top-level directory with input hadron cross-section data
+  //   (search for $GINUKEHADRONDATA or use default location)
+  string data_dir = (gSystem->Getenv("GINUKEHADRONDATA")) ?
```

Comment:
See comment in INukeUtils2025.cxx line 1919 where it is assumed that `$GINUKEHADRONDATA` ends with at `/` slash.   Here that isn't assumed and paths tacked on here all start with a slash.  I'm reviewing these two files in reverse, so what I recommend there is to add the  `/` to the beginning of what's tacked on there.